brave/brave-core
# Use dialog for wayback machine instead of infobar

fix https://github.com/brave/brave-browser/issues/35820

![Screenshot 2024-03-04 151629](https://github.com/brave/brave-core/assets/6786187/14d825dd-9042-4cf5-8941-5159bfcbaa09)
![Screenshot 2024-03-04 151636](https://github.com/brave/brave-core/assets/6786187/2deb6258-d802-4d5e-a430-78ad448feead)

NOTE: button shape is different with design.
It should be handled in a separated issue as it's commonly used button style in all dialogs.

Resolves



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

1. Load https://www.brave.com/bo and check dialog is launched
2. Click check button
3. Check cached page is loaded
4. Load https://www.brave.com/bos and check dialog is launched
5. Click check button
6. Check dialog is updated with not found label
7. Create NTP and relaunch
8. NTP is current active tab dialog is not launched
9. Activate https://www.brave.com/bos tab and check dialog is launched
